nixos/modules/services/databases/couchdb.nix

```diff
@@ -4,20 +4,29 @@ with lib;
 
 let
   cfg = config.services.couchdb;
-  configFile = pkgs.writeText "couchdb.ini"
+  useVersion2 = strings.hasPrefix "2" (builtins.parseDrvName cfg.package.name).version;
```

Comment:
There is `versionOlder` and `versionAtLeast` in lib. (https://github.com/NixOS/nixpkgs/blob/master/lib/strings.nix#L343-L363)